docstore/memdocstore/mem.go

```diff
@@ -352,10 +355,14 @@ func (c *collection) changeRevision(doc map[string]interface{}) {
 }
 
 func (c *collection) checkRevision(arg driver.Document, current map[string]interface{}) error {
-	if current == nil {
-		return nil // no existing document
+	if current == nil || !arg.RevisionOn(c.opts.RevisionField) {
```

Comment:
The RevisionOn check is redundant with the check after GetField below.

I'd leave the RevisionOn check, and instead return an error if GetField returns err != nil.